runtime/lua/vim/treesitter/query.lua

```diff
@@ -552,38 +657,93 @@ local directive_handlers = {
 --- Adds a new predicate to be used in queries
 ---
 ---@param name string Name of the predicate, without leading #
----@param handler function(match:table<string,TSNode>, pattern:string, bufnr:integer, predicate:string[])
+---@param handler function(match: table<integer,TSNode[]>, pattern: integer, source: integer|string, predicate: any[], metadata: table)
```

Comment:
nit:

```suggestion
---@param handler function(match: table<integer,TSNode[]>, pattern: integer, source: integer|string, predicate: any[])
```

Actually there is already a `TSPredicate` type defined below, so we can also reuse it (the existing type annotation was also a duplicate somehow).

Reply:
I see, that's a good point. Reminds myself of [#26721](https://github.com/neovim/neovim/pull/26721/files#diff-c9baf58a11444ecc08b1a8facc69b0bf9a8033d1b3a2771c6ccefd161e987b79R557) (which is outdated, a re-work will follow after this PR) where we had to write something explicitly to make generated docs useful.